server/main.lua

```diff
@@ -13,13 +13,11 @@ local function removeItem(Player, item)
 end
 
 local function checkAndRemoveItem(Player, item, checkRemove)
-	if item then
-		if checkRemove then
-			removeItem(Player, item)
-		end
-		return true
+	if not item then return false end
+	if checkRemove then
```

Comment:
```suggestion
if not checkRemove then return false end
```
Actually, looking at it now, you should make it this instead, otherwise it'll return true when it's not supposed to